InsiteTeamTask/Repositories/DataRepository.cs

```diff
@@ -17,47 +15,136 @@ public List<Attendance> GetAttendanceListFor(int gameNumber)
             List<Ticket> tickets = service.Tickets().ToList();
             List<Product> products = service.Products().ToList();
 
+            List<Attendance> attendanceList = new List<Attendance>();            
+
+            var filteredMembers = members.Where(member => {
+                    var filteredProduct = products.Single(product => product.Id == member.ProductId);
+                
+                    return filteredProduct.GameId == gameNumber;
+                })
+                .Select(member =>
+                {
+                    return new Attendance
+                    {
+                        Barcode = "N/A",
+                        MemberId = member.Id
+                    };
+                });
+
+            attendanceList.AddRange(filteredMembers);
+
+            var filteredTickets = tickets.Where(ticket => {
+                    var filteredProduct = products.Single(product => product.Id == ticket.ProductId);
+
+                    return filteredProduct.GameId == gameNumber;
+                })
+                .Select(ticket =>
+                {
+                    return new Attendance
+                    {
+                        Barcode = ticket.Barcode,
+                        MemberId = 0
+                    };  
+                });
+
+            attendanceList.AddRange(filteredTickets);
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListFor(int gameNumber, int seasonNumber)
+        {
+            MockDataService service = new MockDataService();
+
+            List<Member> members = service.Members().ToList();
+            List<Ticket> tickets = service.Tickets().ToList();
+            List<Product> products = service.Products().ToList();
+
             List<Attendance> attendanceList = new List<Attendance>();
 
-            for(int i = 0; i < members.Count; i++)
-            {
-                attendanceList.Add(new Attendance()
+            var filteredMembers = members.Where(member => {
+                var filteredProduct = products.Single(product => product.Id == member.ProductId);
+
+                return filteredProduct.GameId == gameNumber && filteredProduct.SeasonId == seasonNumber;
+            })
+                .Select(member =>
                 {
-                    Barcode = "N/A",
-                    MemberId = members[i].Id
+                    return new Attendance
+                    {
+                        Barcode = "N/A",
+                        MemberId = member.Id
+                    };
                 });
```

Comment:
Unless you pass in zero to this method as the gameNumber, this step will never return any attendances. Season tickets (`Product` objects with `Type = ProductType.Member`) have access to all games in a season and so all have a GameId of zero.

Secondly, it might make your code more readable and less error-prone to split this up a bit, instead of doing all of it in what is essentially one step:
```
var validProducts = products.Where(p => p.Type == ProductType.Member && p.SeasonId == seasonId);
var matchingMembers = members.Where(m => validProducts.Any(p => p.Id == m.ProductId));
var memberAttendances = matchingMembers.Select(m => new Attendance() { MemberId = m.Id });
```

This removes the `Single()` call you have here, which throws an error if either no matches, or more than one match is found.